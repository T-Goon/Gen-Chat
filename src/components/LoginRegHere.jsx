import React from "react";
import { View, Text, Pressable, StyleSheet } from "react-native";

const LoginRegHere = ({ navigation, msg, location }) => {
    return (
        <View style={{ flexDirection: 'row' }}>
            <Text>{msg}</Text>
            <Pressable onPress={() => { navigation.replace(location); }}>
                <Text style={styles.here}>here</Text>
            </Pressable>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row'
    },
    here: {
        color: 'red',
        borderBottomWidth: 1,
        borderBottomColor: 'red'
    }
});

export default LoginRegHere;