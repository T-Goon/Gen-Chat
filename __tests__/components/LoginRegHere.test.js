import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import LoginRegHere from '../../src/components/LoginRegHere';

describe('<FormInput/ >', () => {
    it('Num Children', () => {
        const replace = jest.fn();
        const loginRegHere = render(<LoginRegHere navigation={{ replace }}
            msg={'Don\'t have an account? Register '} location='Register' />);
        const json = loginRegHere.toJSON();

        expect(json.children.length).toBe(2);
    });

    it('OnPress Prop', () => {
        const replace = jest.fn();
        const { getByText } = render(<LoginRegHere navigation={{ replace }}
            msg={'Don\'t have an account? Login '} location='Login' />);

        fireEvent(getByText('here'), 'press');
        expect(replace).toHaveBeenCalledWith('Login');
    });

    it('Snapshot', () => {
        const replace = jest.fn();
        const loginRegHereJson = render(<LoginRegHere navigation={{ replace }}
            msg={'Don\'t have an account? Too bad. Just click '} location='some place else' />).toJSON();

        expect(loginRegHereJson).toMatchSnapshot();
    });
})