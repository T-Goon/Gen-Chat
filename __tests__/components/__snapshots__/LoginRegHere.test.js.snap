// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput/ > Snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text>
    Don't have an account? Too bad. Just click 
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "borderBottomColor": "red",
          "borderBottomWidth": 1,
          "color": "red",
        }
      }
    >
      here
    </Text>
  </View>
</View>
`;
